### ISSUE TITLE
Examples for the mass??

### ISSUE THREAD

**Issue by unocerobits** [jefffhaynes/ublox#11]
This seems a great library, but I don't idea how I try it.

Can you give us a brief of a simple sample?

Thanks for your great work!

**Comment by jefffhaynes** [comment]
Yeah, it started as a test project and evolved into a bit more over time. I’ll try to write up something about how to use it.

—
If you want to build a ship, don't drum up people to collect wood and don't assign them tasks and work, but rather teach them to long for the endless immensity of the sea.

Antoine de Saint-Exupery
________________________________

**Comment by unocerobits** [comment]
Thanks you. I can help you if you need something with this.

**Comment by RossMelbourne** [comment]
I too would love to see an example of using this library.  Thank you.

**Comment by bikeoid** [comment]
I'll take a stab at sketching out how to interface to the u-blox protocol.

1. Instantiate a ublox.core.Device object, passing it a serial device object that connects to your hardware environment.  
1. Set the event handlers RawPacketHandler, HighNavRatePositionVelocityTimeUpdated, PositionVelocityTimeUpdated and RawDataProductVariantUpdated as required.
1. Call StartListeningAsync, then ConfigurePortAsync and other Configure* as needed to receive the desired UBX messages.

Side note: searching for the term 'BinarySerializer' is just as likely to send to conversations about Microsoft's BinaryFormatter, which is not related to the repo/Nuget package at https://github.com/jefffhaynes/BinarySerializer .

This repo is a fantastic model for detailed interfacing with u-blox devices.   I've upgraded my install to use the latest BinaryFormatter, which also meant updating the CalculateFletcherChecksum class for some changes.   I've added several additional u-blox messages that I needed; not quite all elegant and some just used bytes and ints instead of expanding things out to enums and classes to set bits and fields.
